### PR TITLE
Add enumerated noise_model_type to noise interface

### PR DESCRIPTION
**Pull request by bmhowe23**
This is related to #2235 but will require additional review.

Reference: CUDA-Q Spec Change MR 9 (internal only).
 * Approved by @amccaskey on Oct 1
 * Approved by @bettinaheim on Oct 4

### CODE REVIEW

File: runtime/common/NoiseModel.h [hunk 1]
```diff
@@ -106,6 +117,12 @@ class kraus_channel {
   }
 
 public:
+  /// @brief Noise type enumeration
+  noise_model_type noise_type = noise_model_type::unknown;
+
+  /// @brief Noise probability
+  real probability = 0.0;
```
Comment (amccaskey) [inline runtime/common/NoiseModel.h:124]: Do we envision a need for multiple noise parameters?

Comment (bmhowe23) [reply runtime/common/NoiseModel.h:124]: > Do we envision a need for multiple noise parameters?

Yes, I suppose that could happen for some future noise models. For example, if we wanted to mimic the [PAULI_CHANNEL_1](https://github.com/quantumlib/Stim/blob/main/doc/gates.md#the-pauli_channel_1-instruction) noise option from Stim, then that would need multiple parameters. Do you think we should change `probability` to a vector of doubles?

Comment (amccaskey) [reply runtime/common/NoiseModel.h:124]: Yea I think that would be a good idea, maybe call it parameters

Comment (bmhowe23) [reply runtime/common/NoiseModel.h]: Updated in aa7982853

**Review by 1tnguyen — APPROVED** [review]
LGTM 👍

**Pull request merged**